src/sentry/models/groupsubscription.py

```diff
@@ -34,22 +34,33 @@ class GroupSubscriptionManager(BaseManager):
     def subscribe(
         self,
         group: Group,
-        user: User | RpcUser,
+        subscriber: User | RpcUser | Team,
         reason: int = GroupSubscriptionReason.unknown,
     ) -> bool:
         """
-        Subscribe a user to an issue, but only if the user has not explicitly
+        Subscribe a user or team to an issue, but only if that user or team has not explicitly
         unsubscribed.
         """
+        from sentry.models import Team, User
+
         try:
             with transaction.atomic(router.db_for_write(GroupSubscription)):
-                self.create(
-                    user_id=user.id,
-                    group=group,
-                    project=group.project,
-                    is_active=True,
-                    reason=reason,
-                )
+                if isinstance(subscriber, User) or isinstance(subscriber, RpcUser):
```

Comment:
completely optional, but fyi `isinstance` accepts a Tuple

```suggestion
                if isinstance(subscriber, (RpcUser, User)):
```